okio/src/nonJvmMain/kotlin/okio/internal/Bytes.kt

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okio.internal
+
+import kotlin.math.min
+
+/**
+ * A convenience wrapper around [ByteArray] to allow for
+ * zero allocation operations.
```

Comment:
I suspect this is likely a performance regression because you’re replacing array accesses with polymorphic method calls. Optimizing this kind of thing is not intuitive!

Reply:
(In any case, we should fix in follow-up)

Reply:
... also this says “zero allocation” but it allocates slices all over the place.

Reply:
"zero copy" it is.

I wouldn't be surprised if this is a performance problem. In the event that it is, I will probably abandon the notion of a `chunk` altogether.